### ISSUE TITLE
Nouns appearing as options in multiple choice questions are evaluated when displaying MCQ choices

### ISSUE THREAD

**Issue by mrseanman** [maths/moodle-qtype_stack#1189]
STACK evaluates all nouns when displaying multiple choice question (MCQ) options. My use case is specifically the noun `'diff`, so I will concentrate on that.
When `'diff` is part of a choice in a MCQ, it is evaluated before the tex to display the option is generated. This is a problem when you want to display things like `'diff(f,x)` as options in an MCQ. Instead of displaying the derivative, it evaluates and displays 0. Specific to `diff`, declaring `depends(f,x)` makes no difference to the display of `'diff(f,x)` or `diff(f,x)`. The non-action of `depends` is probably a completely different issue (or intended behavior). I intend to concentrate on the noun issue.

This problem forces the use of `diff(f(x),x)` in MCQs, which would be nice to avoid as this does not typeset as neatly. Also you cannot have, for example `'diff(x^2,x)` as an option in an MCQ. It would display as $2x$.

### Minimal example
_Question variables_
```
depends(f, x);
disp1: 'diff(f,x);
disp2: diff(f,x);

MCQ: [
   [disp1, true],
    /* using the display option makes no difference */
   [2, false, disp2]
];
```
_Question text_
```
These both (correctly) display as \(\frac{\mathrm{d}f}{\mathrm{d}x}\): {@disp1@}, {@disp2@}.
But the MCQ options are both displayed as 0.
[[input:ans1]] [[validation:ans1]]
```
### Notes

- Strangely, if we select the first option, at the PRT, `ans1` is `[diff(f,x)]`, not `[0]`. So it is to do with whatever happens when the tex is generated. The noun is still in there somewhere.
- Use of `noundiff` makes no difference.
- This behavior is similar for checkbox and radio questions.

**Comment by mrseanman** [comment]
This problem appears to extend to all nouns, so have edited title and body text accordingly.

**Comment by sangwinc** [comment]
Thanks for reporting this.  We did have a problem in the past, and I'm sorry about that.  

The logic noun issue ("all nouns" you mention) has a test case which passes and is here
https://github.com/maths/moodle-qtype_stack/blob/master/tests/input_checkbox_test.php#L324

I've tested your example in the "dev" branch (soon to be 4.6.0) and can't reproduce the problem.  

Which version are you using?  (see https://docs.stack-assessment.org/en/Authoring/Author_FAQ/ )

**Comment by mrseanman** [comment]
Thanks for the quick response. The version on my server appears to be 2023060500.

**Comment by sangwinc** [comment]
I've investigated this further.  The display issue is working, but there is another noun-related issue which I'll try to fix in a future release.

**Comment by sangwinc** [comment]
I've been trying to fix this issue for v4.8.0.  The problem is that the attempt is always set to be invalid.

You get the error "Please verify that what you entered was interpreted as expected".
This is triggered by the STACK plugin here
https://github.com/maths/moodle-qtype_stack/blob/master/renderer.php#L143

The state of the question_attempt_step is question_state_invalid.  

What I don't understand is that the STACK question type inputs return a valid/sore result.
